src/packagedcode/golang.py

```diff
@@ -70,10 +71,13 @@ class GolangPackage(models.Package):
 
     @classmethod
     def recognize(cls, location):
-        if fileutils.file_name(location).lower() == 'go.mod':
-            gomod_obj = GoMod()
-            gomod_data = gomod_obj.parse_gomod(location)
+        filename = fileutils.file_name(location).lower()
+        if filename == 'go.mod':
+            gomod_data = GoMod.parse_gomod(location)
             yield build_gomod_package(gomod_data)
+        elif filename == 'go.sum':
+            gosum_objs = go_sum.parse_gosum(location)
```

Comment:
Same as above avoid bland suffix that are not useful such as `_objs`

Reply:
:+1: